// mulaw-js/dist/mulaw-min.js
/*
 mulaw-js
 JavaScript mu-Law codec.
 Copyright (c) 2018 Rafael da Silva Rocha.
 https://github.com/rochars/mulaw-js

 Reference:
 https://github.com/torvalds/linux/blob/master/sound/core/oss/mulaw.c

*/
(function(d){function c(b){if(f[b])return f[b].exports;var a=f[b]={i:b,l:!1,exports:{}};d[b].call(a.exports,a,a.exports,c);a.l=!0;return a.exports}var f={};c.m=d;c.c=f;c.d=function(b,a,g){c.o(b,a)||Object.defineProperty(b,a,{configurable:!1,enumerable:!0,get:g})};c.n=function(b){var a=b&&b.__esModule?function(){return b["default"]}:function(){return b};c.d(a,"a",a);return a};c.o=function(b,a){return Object.prototype.hasOwnProperty.call(b,a)};c.p="";return c(c.s=0)})([function(d,c){function f(a){if(0>
a){a=132-a;var b=127}else a+=132,b=255;32767<a&&(a=32767);var c=0;var e=a>>7;e&240&&(e>>=4,c+=4);e&12&&(e>>=2,c+=2);e&2&&(c+=1);e=c;return(e<<4|a>>e+3&15)^b}function b(a){a=~a;var b=((a&15)<<3)+132<<((a&112)>>4);return a&128?132-b:b-132}d.exports.encodeSample=f;d.exports.decodeSample=b;d.exports.encode=function(a){for(var b=[],c=0;c<a.length;c++)b.push(f(a[c]));return b};d.exports.decode=function(a){for(var c=[],d=0;d<a.length;d++)c.push(b(a[d]));return c}}]);
